deepchecks/recommender/ranking.py

```diff
@@ -7,705 +7,423 @@
 
-    return np.mean(ars)
+    return np.mean(f1_list)
 
 
-def hits_at_k(relevant_items: np.array, recommendation: np.array, k=10):
+def reciprocal_rank(relevant_item,recommendation: Union[List, np.array, None]) -> float:
     """
-    Calculate the number of hits of relevant items in a ranked list HITS@K.
+    Calculates the reciprocal rank for a given recommendation.
 
     Args:
-        relevant_items (array-like): An 1 x N array of relevant items.
-        recommendation (array-like): An 1 x N array of predicted arrays
-        k (int): the number of items considered in the predicted list
+    - relevant_item: integer representing the relevant item
+    - recommendation: list or numpy array of recommended items
+
     Returns:
-        HITS@K (float):  The number of relevant items in the first k items of a prediction.
+    - float: reciprocal rank of the relevant item in the recommendation
     """
-    if len(recommendation) > k:
-        recommendation = recommendation[:k]
-
-    hits = 0.0
-    for i, item in enumerate(recommendation):
-        if item in relevant_items and item not in recommendation[:i]:
-            hits += 1.0
-
-    if len(recommendation) == 0:
+    if relevant_item is None or len(recommendation) == 0:
         return 0.0
-    return hits / len(recommendation)
+    for rank, rec_item in enumerate(recommendation):
+        if relevant_item == rec_item:
+            return 1/(rank+1)
+        return 0
 
 
-def spearmans_rho(relevant_items: np.array, recommendation: np.array):
+def mean_reciprocal_rank(relevant_items: List, recommendations: List[List]) -> float:
     """
-    Calculate the Spearman's rank correlation coefficient (Spearman's rho) between two lists.
-
+    Calculates the mean reciprocal rank (MRR) of the recommendations given for the relevant items.
+    
     Args:
-        relevant_items (array-like): An 1 x N array of items.
-        recommendation (array-like):  An 1 x N array of items.
+        relevant_items (List): A list of relevant items for each user or item.
+        recommendations (List[List]): A list of recommended items for each user or item.
+    
     Returns:
-        (float): Spearman's rho.
-        p-value (float): two-sided p-value for null hypothesis that both predicted are uncorrelated.
+        float: The mean reciprocal rank (MRR) of the recommendations.
     """
-    return stats.spearmanr(relevant_items, recommendation)
-
+    reciprocal_ranks = [reciprocal_rank(item, recommendation) \
+                        for item, recommendation \
+                        in zip(relevant_items, recommendations)]
 
-def kendall_tau(relevant_items: np.array, recommendation: np.array):
-    """
-    Calculate the Kendall's tau, measuring the correspondence between two lists.
+    return np.mean(reciprocal_ranks)
 
-    Args:
-        relevant_items (array-like): An 1 x N array of items.
-        recommendation (array-like):  An 1 x N array of items.
+def dcg(y_true : np.ndarray,
```

Comment:
We must have name consistency for the current metric calculation method to work. If previously we called y_true relevant_items and y_pred recommendations, we should keep these names in all metrics.